test/core/end2end/end2end_test_main.cc

```diff
@@ -0,0 +1,1044 @@
+    grpc_server_credentials* ssl_creds = grpc_ssl_server_credentials_create(
+        nullptr, &pem_key_cert_pair, 1, 0, nullptr);
+    grpc_slice_unref(cert_slice);
+    grpc_slice_unref(key_slice);
+    if (args.Contains(FAIL_AUTH_CHECK_SERVER_ARG_NAME)) {
+      grpc_auth_metadata_processor processor = {ProcessAuthFailure, nullptr,
+                                                nullptr};
+      grpc_server_credentials_set_auth_metadata_processor(ssl_creds, processor);
+    }
+
+    auto* server = grpc_server_create(args.ToC().get(), nullptr);
+    grpc_server_register_completion_queue(server, cq(), nullptr);
+    GPR_ASSERT(grpc_server_add_http2_port(
+        server, grpc_end2end_proxy_get_server_port(proxy_), ssl_creds));
+    grpc_server_credentials_release(ssl_creds);
+    grpc_server_start(server);
+    return server;
+  }
+
+  grpc_channel* MakeClient(const ChannelArgs& args) override {
+    grpc_channel_credentials* ssl_creds =
+        grpc_ssl_credentials_create(nullptr, nullptr, nullptr, nullptr);
+    auto* client = grpc_channel_create(
+        grpc_end2end_proxy_get_client_target(proxy_), ssl_creds,
+        args.Set(GRPC_SSL_TARGET_NAME_OVERRIDE_ARG, "foo.test.google.fr")
+            .ToC()
+            .get());
+    GPR_ASSERT(client != nullptr);
+    grpc_channel_credentials_release(ssl_creds);
+    return client;
+  }
+  const grpc_end2end_proxy_def proxy_def_ = {CreateProxyServer,
+                                             CreateProxyClient};
+  grpc_end2end_proxy* proxy_;
+};
+
+class FixtureWithTracing final : public CoreTestFixture {
+ public:
+  explicit FixtureWithTracing(std::unique_ptr<CoreTestFixture> fixture)
+      : fixture_(std::move(fixture)) {
+    // g_fixture_slowdown_factor = 10;
+    EXPECT_FALSE(grpc_tracer_set_enabled("doesnt-exist", 0));
+    EXPECT_TRUE(grpc_tracer_set_enabled("http", 1));
+    EXPECT_TRUE(grpc_tracer_set_enabled("all", 1));
+  }
+  ~FixtureWithTracing() override {
+    saved_trace_flags_.Restore();
+    // g_fixture_slowdown_factor = 1;
+  }
+
+  grpc_server* MakeServer(const ChannelArgs& args) override {
+    return fixture_->MakeServer(args);
+  }
+
+  grpc_channel* MakeClient(const ChannelArgs& args) override {
+    return fixture_->MakeClient(args);
+  }
+
+ private:
+  SavedTraceFlags saved_trace_flags_;
+  std::unique_ptr<CoreTestFixture> fixture_;
+};
+
+#ifdef GRPC_POSIX_WAKEUP_FD
+class InsecureFixtureWithPipeForWakeupFd : public InsecureFixture {
+ public:
+  InsecureFixtureWithPipeForWakeupFd()
+      : old_value_(std::exchange(grpc_allow_specialized_wakeup_fd, 0)) {}
+
+  ~InsecureFixtureWithPipeForWakeupFd() override {
+    grpc_allow_specialized_wakeup_fd = old_value_;
+  }
+
+ private:
+  const int old_value_;
+};
+#endif
+
+const char* NameFromConfig(
+    const ::testing::TestParamInfo<const CoreTestConfiguration*>& config) {
+  return config.param->name;
+}
+
+const NoDestruct<std::vector<CoreTestConfiguration>> all_configs{std::vector<
```

Comment:
Please dump out the full matrix of fixtures x tests with both the old and new code, and make sure we're not accidentally missing anything.

Reply:
https://gist.github.com/ctiller/919a358a71c6b0371bfc828a9a445f76

I'll try to do some data processing on this, but it's non trivial (a lot of tests in the old system become N individual tests in the new)